synapse/rest/client/v2_alpha/receipts.py

```diff
@@ -42,10 +43,24 @@ async def on_POST(self, request, room_id, receipt_type, event_id):
         if receipt_type != "m.read":
             raise SynapseError(400, "Receipt type must be 'm.read'")
 
+        body = parse_json_object_from_request(request)
+        hidden = body.get("org.matrix.msc2285.hidden", False)
+
+        if not isinstance(hidden, bool):
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST,
```

Comment:
I'd just use the integer 400 here, for consistency with the rest of the code base.